runtime/signals_nat.c

```diff
@@ -306,3 +329,17 @@ CAMLexport int caml_setup_stack_overflow_detection(void)
   return 0;
 #endif
 }
+
+CAMLexport int caml_stop_stack_overflow_detection(void)
+{
+#ifdef HAS_STACK_OVERFLOW_DETECTION
+  stack_t oldstk, stk;
+  stk.ss_flags = SS_DISABLE;
+  if (sigaltstack(&stk, &oldstk) == -1) return -1;
+  /* If caml_setup_stack_overflow_detection failed, we are not using
+     an alternate signal stack.  SS_DISABLE will be set in oldstk,
+     and there is nothing to free in this case. */
+  if (! (oldstk.ss_flags & SS_DISABLE)) free(oldstk.ss_sp);
+#endif
+  return 0;
```

Comment:
It seems to me that we could keep the same exact code as the one in 4.14 rather than merging the two `return 0;`?

Reply:
Done - I was possibly diff'ing in the other direction by this point!